utils/nctl/sh/scenarios-upgrades/upgrade_scenario_01.sh

```diff
@@ -0,0 +1,240 @@
+# 2. Execute some deploys to populate global state a little
+# 3. Upgrade all running nodes to v2
+# 4. Assert v2 nodes run & the chain advances (new blocks are generated)
+# 5. Start passive nodes.  The launcher should cause the v1 node to run, it should exit and the v2 node should then catch up.
+# 6. Assert passive nodes are now running.
+
+# ----------------------------------------------------------------
+# Imports.
+# ----------------------------------------------------------------
+
+source "$NCTL/sh/utils/main.sh"
+source "$NCTL/sh/node/svc_$NCTL_DAEMON_TYPE".sh
+
+# ----------------------------------------------------------------
+# ENTRY POINT
+# ----------------------------------------------------------------
+
+unset _STAGE_ID
+
+for ARGUMENT in "$@"
+do
+    KEY=$(echo "$ARGUMENT" | cut -f1 -d=)
+    VALUE=$(echo "$ARGUMENT" | cut -f2 -d=)
+    case "$KEY" in
+        stage) _STAGE_ID=${VALUE} ;;
+        *)
+    esac
+done
+
+_STAGE_ID="${_STAGE_ID:-1}"
+
+# ----------------------------------------------------------------
+# MAIN
+# ----------------------------------------------------------------
+
+# Main entry point.
+function _main()
+{
+    local STAGE_ID=${1}
+
+    if [ ! -d $(get_path_to_stage "$STAGE_ID") ]; then
+        log "ERROR :: stage $STAGE_ID has not been built - cannot run scenario"
+        exit 1    
+    fi
+
+    _step_01 "$STAGE_ID"
+    _step_02
+    _step_03
+    _step_04
+    _step_05 "$STAGE_ID"
+    _step_06
+    _step_07
+    _step_08
+    _step_09
+    _step_10
+}
+
+# Step 01: Start network from pre-built stage.
+function _step_01() 
+{
+    local STAGE_ID=${1}
+
+    log_step 1 "starting network from stage $STAGE_ID"
+
+    source "$NCTL/sh/assets/setup_from_stage.sh" stage="$STAGE_ID"
+    source "$NCTL/sh/node/start.sh" node=all
+}
+
+# Step 02: Await era-id >= 1.
+function _step_02() 
+{
+    log_step 2 "awaiting genesis era completion"
+
+    sleep 30.0
+    await_until_era_n 1
+}
+
+# Step 03: Populate global state -> native + wasm transfers.
+function _step_03() 
+{
+    log_step 3 "dispatching deploys to populate global state"
+
+    log "... ... 100 native transfers"
+    source "$NCTL/sh/contracts-transfers/do_dispatch_native.sh" \
+        transfers=100 interval=0.0 verbose=false
+
+    log "... ... 100 wasm transfers"
+    source "$NCTL/sh/contracts-transfers/do_dispatch_wasm.sh" \
+        transfers=100 interval=0.0 verbose=false
+}
+
+# Step 04: Await era-id += 1.
+function _step_04() 
+{
+    log_step 4 "awaiting next era"
+
+    await_n_eras 1
+}
+
+# Step 05: Upgrade network from stage.
+function _step_05() 
+{
+    local STAGE_ID=${1}
+
+    log_step 5 "upgrading network from stage ($STAGE_ID)"
+
+    source "$NCTL/sh/assets/upgrade_from_stage.sh" stage="$STAGE_ID"
+    sleep 10.0
+}
+
+# Step 06: Assert chain is progressing at all nodes.
+function _step_06() 
+{
+    local HEIGHT_1
+    local HEIGHT_2
+    local NODE_ID
+
+    log_step 6 "asserting node upgrades"
```

Comment:
I'd change this message since this function in stall related and doesn't currently check if they upgraded.

Or probably better to do the upgrade assertion here as well.